qip-0007.md

```diff
@@ -57,43 +55,39 @@ The value field of a transaction output is constrained to 1 of 16 denominations,
 | 15           | 100000 Qi    |
 
 ### Transaction Structure
-Every transaction must reference a set of UTXOs as inputs to be spent, and create new transaction outputs to be added to the UTXO set. Transaction inputs will be referenced by the transaction hash which minted the UTXO and the output index of that transaction.
-
-For example, consider a previous transaction with hash `0x11223344` which created 4 new outputs. For a subsequent transaction to spend the 2nd output from that transaction, it would create the following input:
-```
-{
-  txhash: "0x11223344",  // Hash of the transaction which created the outputs
-  index: 1,              // Output index, counting from 0
-  signature: 0x...,      // Signature omitted from this example
-}
-```
+Every transaction must reference a set of unspent transaction outputs as inputs to be spent, and create new transaction outputs to be added to the UTXO set. Transaction inputs will be referenced by the hash of the block which minted them and the index of the output in that block. Additionally, each transaction must contain a valid aggregate signature from each UTXO key.
 
-Additionally, each input must be accompanied by a valid signature from the UTXO pubkey. The signature must be a valid ECDSA SECP256K1 signature of the transaction hash. Putting it all together, transactions will contain the following data, RLP encoded:
+Putting it all together, transactions will contain the following data, RLP encoded:
 ```
-{
+transaction: {
   inputs: [
     {
-      txhash: []byte,
-      index: uint8,
-      signature: []byte,
+      blockhash: []byte, // block which minted the output
+      index: uint8,      // output index of that block
+      pubkey: []byte,    // public key corresponding to the output's address
     },
     ...
   ],
   outputs: [
     {
-      version: uint8,
-      denomination: uint8,
-      address: []byte,
+      denomination: uint8, // which denomination of coins this output represents
+      address: []byte,     // owner's address
     },
     ...
   ],
+  signature: []byte, // MuSig aggregate signature
```

Comment:
technically this is just a multisig schnorr. How or why it is generated is independent.

Reply:
good point